operator/internal/component/podgangset/hpa/hpa.go

```diff
@@ -138,10 +138,17 @@ func (r _resource) computeExpectedHPAs(pgs *grovecorev1alpha1.PodGangSet) []hpaI
 				Namespace: pgs.Namespace,
 				Name:      pclqFQN,
 			}
+			// Use the initial replicas count as minimum for PodClique HPAs
+			minReplicas := pclqTemplateSpec.Spec.Replicas
+			if pclqTemplateSpec.Spec.ScaleConfig.MinReplicas != nil {
+				minReplicas = *pclqTemplateSpec.Spec.ScaleConfig.MinReplicas
+			}
```

Comment:
This can be simplified significantly:

```suggestion
			// Use the initial replicas count as minimum for PodClique HPAs
            minReplicas := ptr.Deref(pclqtemplateSpec.Spec.ScaleConfig.MinReplicas, pclqTemplateSpec.Spec.Replicas)
```
from package `k8s.io/utils/ptr`.

Also, I am puzzled why we are currently maintaning an additional field just for the `MinReplicas`, when this information will *already* be present in the `ScaleConfig` that is in `pclqTemplateSpec.Spec.ScaleConfig`? We might have missed this before; but I don't see any reason to maintain this seperately. `MaxReplicas` is only present in `ScaleConfig`, and I see no reason to treat `MinReplicas` any differently.

Reply:
I updated the logic, could you please double check if that makes sense ?

Reply:
ok, I removed this logic completely from HPA and moved it to the default webhook as per our slack huddle